pubsub/google/cloud/pubsub_v1/subscriber/_consumer.py

```diff
@@ -244,10 +244,9 @@ def _blocking_consume(self):
                 self.stop_consuming()
             except Exception as exc:
                 try:
-                    self._policy.on_exception(exc)
+                    self.active = self._policy.on_exception(exc)
```

Comment:
Active gets set to `False` here, but what actually makes the consumer exit? From what I can tell, active will be set to false but the next loop here will just re-open the stream.

Reply:
I am also hazy on this. See my [note][1], AFAICT `.close()` (which happens [here][2]) is what cleans up (it's the only thing that can stop `Consumer._blocking_consume` since `Consumer.stop_consuming` is the only thing that calls `self._exiting.set()`).

[1]: https://github.com/GoogleCloudPlatform/google-cloud-python/pull/4472/files#r153624395
[2]: https://github.com/GoogleCloudPlatform/google-cloud-python/blob/56fc8843728e606523f035763f93d8d7bd4aeb90/pubsub/google/cloud/pubsub_v1/subscriber/policy/base.py#L289-L291

Reply:
Yeah, it seems to me that `active` is just an external signal and doesn't impact flow at all within this class (maybe that's worth adding a comment).

ISTM that an unrecoverable error here should have the same impact as `stop_consuming` below.

Reply:
I _think_ that is the best answer.

Reply:
I dropped the `try/except` and just always set `active`. I am going to add another unit test which is OK with one exception but not with another.